src/Microsoft.ApiDesignGuidelines.Analyzers/Core/IdentifiersShouldNotContainTypeNames.cs

```diff
@@ -94,6 +94,13 @@ public override void Initialize(AnalysisContext analysisContext)
             // Analyze properties and methods, and their parameters.
             analysisContext.RegisterSymbolAction(symbolContext =>
             {
+                if ((symbolContext.Symbol as IMethodSymbol).IsEqualsOverride())
```

Comment:
Yes, `IsEqualsOverride` contains a null check. It must have, otherwise this analyzer would always fail on `IPropertySymbol`, which flows through the same code path.

To me, using the `as` operator already indicates that the conversion may fail. Otherwise a cast would be the right approach. 

I'm not a fan to adding extra code that does not change meaning, but I can see how passing in `null` to `IsEqualsOverride` would return `false` is not explicit, so I'll change:

```csharp
if ((symbolContext.Symbol as IMethodSymbol).IsEqualsOverride())
```

to the following:

```csharp
if ((symbolContext.Symbol as IMethodSymbol)?.IsEqualsOverride() == true)
```

if you want me to. Please let me know.